### ISSUE TITLE
Invalid error message given when using fromJsonWithType, fromJsonStringWithType on JSON compatible types

### ISSUE THREAD

**Issue by pubudu91** [ballerina-platform/ballerina-lang#24247]
Consider the following snippet:
```ballerina
string s = "foobar";
int x2 = checkpanic s.fromJsonWithType(int);
io:println(x2);
```
The following compile error is given for the above:
```
error: .::test2.bal:33:25: incompatible types: expected '(int|error)', found '(int|error)'
```
IINM, this should be allowed at compile time, but should panic at runtime.

**Comment by pubudu91** [comment]
The following has the same problem:
```ballerina
int i = 20;
string x2 = checkpanic i.fromJsonWithType(string);
```

**Comment by pubudu91** [comment]
And in `fromJsonStringWithType()` as well:
```ballerina
string jSon = "10";
string val = checkpanic jSon.fromJsonStringWithType(string);
```

**Comment by pubudu91** [comment]
Doesn't seem to work without checkpanic either.
```ballerina
string s = "foobar";
int|error zz = s.fromJsonWithType(int);
```

**Comment by dulajdilshan** [comment]
This issue is fixed with [#24057](https://github.com/ballerina-platform/ballerina-lang/pull/24057)
